backend/players/models.py

```diff
@@ -2,19 +2,110 @@
+    avatar = models.ImageField(upload_to='avatars/', default='fallback.png', blank=True) # Profile picture
+    created_at = models.DateTimeField(default=timezone.now) # Date of account creation
+    updated_at = models.DateTimeField(auto_now=True) # Date of last account update
 
     def __str__(self):
-        return self.display_name
+        return f"Player: {self.display_name}"
 
 # Use signal to remove corresponding user on player deletion
 @receiver(post_delete, sender=Player)
 def delete_user_with_player(sender, instance, **kwargs):
     # Delete the associated user when the player is deleted
     if instance.user:
         instance.user.delete()
+
+
+class Tournament(models.Model):
+    name = models.CharField(max_length=100) # Name of the tournament
+    status = models.CharField(max_length=20, choices=[
+        ('scheduled', 'Scheduled'),
+        ('in_progress', 'In Progress'),
+        ('completed', 'Completed')
+    ], default='scheduled') # Status of the tournament
+    winner = models.ForeignKey(Player,
+                               on_delete=models.SET('Unknown'),
+                               null=True,
+                               blank=True,
+                               related_name='won_tournaments') # Winner of the tournament
+    created_at = models.DateTimeField(auto_now_add=True) # Date of tournament creation
+
+    def __str__(self):
+        return f"Tournament: {self.name} - Status: {self.status}"
+
+# Association table for players and tournaments
+class TournamentParticipant(models.Model):
+    tournament = models.ForeignKey(Tournament,
+                                   on_delete=models.CASCADE,
+                                   related_name='participants') # Tournament the player is participating in
+    player = models.ForeignKey(Player, on_delete=models.CASCADE) # Player participating in the tournament
+    order = models.IntegerField() # Order of the player in the tournament (For seeding purposes)
+
+    class Meta:
+        unique_together = ('tournament', 'player')
+
+    def __str__(self):
+        return f"{self.player.display_name} in {self.tournament.name}"
+
+
+class Match(models.Model):
+    tournament = models.ForeignKey(Tournament,
+                                   on_delete=models.CASCADE,
+                                   related_name='matches',
+                                   null=True,
+                                   blank=True) # Tournament the match is part of
+    player1 = models.ForeignKey(Player,
+                                on_delete=models.CASCADE,
+                                related_name='player1_matches') # First player in the match
+    player2 = models.ForeignKey(Player,
+                                on_delete=models.CASCADE,
+                                related_name='player2_matches') # Second player in the match
+    round_of = models.IntegerField(null=True, blank=True) # Tournament round (round of 16 (1/8), 8 (1/4), 4 (1/2), 2 (final))
```

Comment:
Will we ever have round of 16, 8?
I think it will be only 4 (semi) and 2 (final). Otherwise, we would need to login 8 and 16 people or even more from one browser window. We should keep it simple. 

We can consider using `CharField` with `choices` and stick to semifinals and final.
```suggestion
        ROUND_CHOICES = [
                ('SF', 'Semifinals'),
                ('F', 'Final'),
        ]
        round_stage = models.CharField(max_length=2, choices=ROUND_CHOICES, null=True, blank=True)
```